lib/typeUtil.js

```diff
@@ -90,28 +90,40 @@ function valueToObject(value) {
   }
 }
 
+function objectToType(obj) {
+  var objectType = Object.keys(obj)
+  if (objectType.length != 1) {
+    throw new Error('Expected only one key from Amazon object')
+  }
+
+  return objectType[0]
+}
+
 /**
  * Convert a Dynamo AttributeValue value to a javascript primitive value
  *
  * @param {Object} obj Dynamo AttributeValue value.
  * @return {string|number|Array} a javascript primitive value
  */
-function _objectToValue(obj) {
-  if (obj.hasOwnProperty('SS')) {
-    return obj.SS
-  } else if (obj.hasOwnProperty('NS')) {
-    var numArray = []
-    for (var i = 0; i < obj.NS.length; i++) {
-      numArray.push(Number(obj.NS[i]))
-    }
-    return numArray
+function objectToValue(obj) {
+  switch (objectToType(obj)) {
+    case 'SS':
+      return obj['SS']
+    case 'S':
+      return obj['S']
+    case 'NS':
+      return obj['NS'].map(function (num) { return Number(num) })
```

Comment:
For SS and NS we're given back a list of strings.  SS needs to return a list of strings (already good) and NS needs to return a list of integers (needs the `.map` code).

(Note I just roughly copied the code that was here previously).